src/bvh/collision_object.hpp

```diff
@@ -254,6 +252,8 @@ namespace bvh
         } );
     }
 
+  private:
```

Comment:
What is the reason these are moved? Same as below

Reply:
`nvcc` doesn't like `update_snapshots` with `private` (or `protected`):
```
/distBVH/src/bvh/collision_object.hpp(237): error: The enclosing parent function ("update_snapshots") for an extended __host__ __device__ lambda cannot have private or protected access within its class
```

Reply:
Ugh.. I forgot about that. Ok this is fine for now, maybe in the future we can try and work around it since these should be private